Sources/JSON/Array+JSONConvertible.swift

```diff
@@ -0,0 +1,35 @@
+extension Array: JSONConvertible {
+    public init(json: JSON) throws {
+        guard Element.self is JSONInitializable.Type else {
```

Comment:
@tanner0101 guard let was crashing for me occasionally on linux, this is a work around adapted from Node.